npm/src/filechecks/index.ts

```diff
@@ -0,0 +1,46 @@
+import { GraphqlClient } from "../graphql"
+import {
+  getConsignmentData,
+  updateProgressBar,
+  IFileCheckProcessed,
+  getConsignmentId
+} from "./file-check-processing"
+export class FileChecks {
+  client: GraphqlClient
+
+  constructor(client: GraphqlClient) {
+    this.client = client
+  }
+
+  dataCallback: (
+    fileChecksProcessed: IFileCheckProcessed | null
+  ) => void = fileCheckProcessed => {
+    if (fileCheckProcessed) {
+      const {
+        antivirusProcessed,
+        checksumProcessed,
+        totalFiles
+      } = fileCheckProcessed
+
+      updateProgressBar(
+        antivirusProcessed,
+        totalFiles,
+        "#av-metadata-progress-bar"
+      )
+      updateProgressBar(checksumProcessed, totalFiles, "#checksum-progress-bar")
+
+      if (antivirusProcessed == totalFiles && checksumProcessed == totalFiles) {
+        const location = `${
+          window.location.origin
+        }/consignment/${getConsignmentId()}/records-results`
+        window.location.href = location
+      }
+    }
+  }
+
+  updateFileCheckProgress() {
+    const interval = setInterval(() => {
+      getConsignmentData(this.client, this.dataCallback)
+    }, 2000)
```

Comment:
I ran this locally and was thinking is it possible for us to update the processing bars more often than every 2 seconds so it moves similarly to the upload bar? Though this could just be because I tested it with quite a small folder!

Reply:
It's hard to tell what's best. The upload bar updates when the file checks are done or the s3 upload completes rather than being run on a timer like this one so it's difficult to compare. We have to find the right balance between not being too slow on the page and not hammering the api with requests. I think 2 seconds is ok for now and we can change it if the user testing shows it needs to be longer or shorter.

Reply:
Good point! :+1: